nncf/openvino/graph/model_builder.py

```diff
@@ -0,0 +1,222 @@
+        Since these nodes are constants and do not require any nodes as inputs, cloned nodes will not be used.*
+            graph_nodes = [convolution, parameter, fake quantize, oh, ol, ih, il]
+            clone_nodes = [c, il, ih, ol, oh]
+
+        *creating fake quantize node.
+        This node requires to have input values in a specific order.
+        All previous nodes will be connected/used for fake quantize, from left to right.*
+            graph_nodes = [convolution, parameter]
+            clone_nodes = [f]
+
+        *creating parameter node.
+        In this step, the list of parameters will also be filled out with the new node.*
+            graph_nodes = [convolution]
+            clone_nodes = [f, parameter]
+
+        *creating convolution node.
+        This node also requires to have inputs in a specific order.
+        All previous nodes will be connected/used for convolution, from left to right. Also,
+        the outputs verification step will show here that one of the convolution outputs is in the output_ids list.
+        This means that the Result node would be created and placed into the results list.*
+            graph_nodes = []
+            clone_nodes = [convolution]
+
+        The last step is to create a subgraph model based on the parameters & results lists.
+
+        :param input_ids: List of the ids specified in algorithm.
+        :param output_ids: List of the ids specified in algorithm.
+        :param node_mapping: Original nodes mapping.
+        :return: Builded ov.Model based on parameters.
+        """
+
+        parameters, results = [], []
+        clone_nodes = deque()
+
+        # Collecting nodes that declares the graph.
+        graph_nodes = self._collect_graph_nodes(input_ids, output_ids, node_mapping)
+
+        while graph_nodes:
+            graph_node = graph_nodes.pop()
+            node_type = graph_node.get_type_name()
+            node_name = graph_node.get_friendly_name()
+
+            # To create the new OpenVINO nodes, we need to provide all possible layer attributes.
+            attrs = graph_node.get_attributes()
+            attrs["name"] = node_name
+
+            if node_type == "Constant":
+                # Constants creation is apart due to specific behavior.
+                clone_node = OVModelTransformer._create_constant(
+                    graph_node.get_data(), dtype=graph_node.get_element_type(), name=attrs["name"]
+                )
+            elif node_type == "Parameter":
+                # We've created Parameter nodes on the previous step.
+                clone_node = graph_node
+                parameters.append(clone_node)
+            else:
+                # We have to have args as the inputs since all of them are nodes and are required to be as input.
+                args = [clone_nodes.popleft() for _ in graph_node.inputs()]
+
+                clone_node = self._node_factory.create(node_type, args, attrs)
```

Comment:
Currently, `NodeFactory` will create nodes without names since naming is done using a decorator (see https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/openvino/utils/decorators.py#L25-L34).
It shouldn't be a problem, since the FastBC algorithm utilizes `Parameter` and `Result` nodes for statistics. But for BiasCorrection algorithm utilizes the names.
The current approach may be changed by adding the corresponding constructor calls into _metatypes,_ for example, for each layer.